src/factoid.rs

```diff
@@ -76,14 +76,31 @@ impl FactoidKnowledge for Brain {
     }
 }
 
-pub fn is_awoo(s: &String) -> bool {
-    let lower_s = s.to_ascii_lowercase();
-    if !lower_s.starts_with("awoo") {
+// e.g. awoo -> awooooo or meow -> meoooow
+fn is_extension(base: &String, candidate: &String)  -> bool {
+    if base.len() == 0 && candidate.len() != 0 {
         return false;
+    } else if base.len() == 0  && candidate.len() == 0 {
+        return true;
     }
+    
+    // blame ixi
+    let h = base.chars().next().unwrap(); // Safe, we checked it wasn't empty above
```

Comment:
aside anything else, these lines are a bit nicer if you yoink out `base.chars()` and reuse the iterator:

```
    let mut base_chars = base.chars();

    // do not blame ixi
    let h = base_chars.next().unwrap(); // Safe, we checked it wasn't empty above
    let c = candidate.chars().skip_while(|x| x == &h);
    is_extension(&base_chars.collect::<String>(), &c.collect::<String>())
```